src/realm/util/file.cpp

```diff
@@ -562,7 +563,8 @@ void File::close() noexcept
 
     if (m_fd < 0)
         return;
-    unlock();
+    if (m_have_lock)
```

Comment:
Yes, but bear in mind that is very likely that since the system call for unlocking the file fails on certain circumstances, the same system call will eventually fail also when locking the file. So an exception will be thrown either way.